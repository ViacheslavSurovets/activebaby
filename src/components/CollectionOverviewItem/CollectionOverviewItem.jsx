import React, { useState } from 'react';
import {
  CategoryItemImageLink,
  CategoryItem,
  CategoryItemCustomButtonWrapper,
  CategoryItemInfoWrapper,
  CategoryItemPriceText,
  CategoryItemText,
} from './styles';
import { CustomButton } from '@components/CustomButton';
import { ShopCartIcon } from '@icons/index';
import { useTranslation } from 'react-i18next';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { addItem } from '@redux/cart/cart.actions';
import { addItemToPreviewComponent, showPortalPreviewItem } from '@redux/previewItem/previewItem.actions';


const CollectionOverviewItem = ( { item, info, addItem, addPreviewItem, showPortalPreviewItem } ) => {
  const { t } = useTranslation ();

  const { images } = item;

  const handleClickImage = async () => {
    await addPreviewItem ( item, info );
    await showPortalPreviewItem();
  };

  return (
    <CategoryItem>
      <CategoryItemImageLink onClick={ handleClickImage } background={ images[ 0 ] } to={ '#' } />
      <CategoryItemInfoWrapper>
        <CategoryItemText><CategoryItemPriceText>{ t ( 'shopItem.price' ) }:</CategoryItemPriceText> { item.price } BYN</CategoryItemText>
        <CategoryItemCustomButtonWrapper>
          <CustomButton onClick={ () => addItem ( item ) } orangeSoft width={ '4rem' }
                        height={ '3rem' }><ShopCartIcon /></CustomButton>
        </CategoryItemCustomButtonWrapper>
      </CategoryItemInfoWrapper>
    </CategoryItem>
  );
};


const mapDispatchToProps = dispatch => ({
  addItem: item => dispatch ( addItem ( item ) ),
  addPreviewItem: ( item, info ) => dispatch ( addItemToPreviewComponent ( item, info ) ),
  showPortalPreviewItem: () => dispatch ( showPortalPreviewItem () )
});

export default connect ( null, mapDispatchToProps ) ( CollectionOverviewItem );


CollectionOverviewItem.propTypes = {
  item: PropTypes.object,
  info: PropTypes.string,
  addItem: PropTypes.func,
  addPreviewItem: PropTypes.func,
  showPortalPreviewItem: PropTypes.func
};
