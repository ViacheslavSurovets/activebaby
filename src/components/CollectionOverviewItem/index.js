export { default as CollectionOverviewItem } from './CollectionOverviewItem';
