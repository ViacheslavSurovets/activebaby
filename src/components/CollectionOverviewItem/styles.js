import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { commonStyleTextTwoRem, commonLinkStyle } from '@commonStyles';


export const CategoryItem = styled.div`
  display: flex;
  flex-direction: column;
  width: 25rem;
  height: 30rem;
  margin: 1rem;
  transition: all 1s;

  background-color: ghostwhite;
  &:hover{
    transform: scale(1.05);
    transition: all 1s;
    box-shadow: 4px 4px 4px -3px ${ ( { theme } ) => theme.black };
  }
`;

export const CategoryItemImageLink = styled ( Link )`
  ${ commonLinkStyle };
  background-image: url(${ props => props.background });
  background-size: 100% 100%;
  flex: 1;
  margin: .5rem;
`;


export const CategoryItemText = styled.span`
  ${ commonStyleTextTwoRem };
  font-weight: bold;
`;

export const CategoryItemPriceText = styled.span`
  font-size: 1.6rem;
  font-weight: lighter;
`;

export const CategoryItemInfoWrapper = styled.div`
  display: flex;
  align-items: center; 
  justify-content: space-between;
  flex-wrap: wrap;
`;

export const CategoryItemCustomButtonWrapper = styled.div`

  &:hover {
   svg {
        fill: ${ ( { theme } ) => theme.header.cartColor };
      };
  }
`;
