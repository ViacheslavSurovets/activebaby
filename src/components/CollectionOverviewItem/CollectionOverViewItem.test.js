import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import CollectionOverviewItem from './CollectionOverviewItem';
import configureMockStore  from 'redux-mock-store';


const mockStore = configureMockStore();
const store = mockStore({});

describe('CollectionOverviewItem', ()=> {
  it('CollectionOverviewItem Snapshot', ()=> {
    expect ( shallow (
      <Provider store={store}>
        <CollectionOverviewItem  />
      </Provider>
    ) ).toMatchSnapshot();
  });
});
