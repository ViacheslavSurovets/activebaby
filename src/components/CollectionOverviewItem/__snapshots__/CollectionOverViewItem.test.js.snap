// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionOverviewItem CollectionOverviewItem Snapshot 1`] = `ShallowWrapper {}`;
